soh/src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -17,6 +20,638 @@
+        if (it == item || (item == ITEM_BOTTLE && it >= ITEM_BOTTLE && it <= ITEM_POE)) {
+            return 1;
+        }
+    }
+
+    if (item >= ITEM_SONG_MINUET && item <= ITEM_SONG_STORMS) {
+        return (Save_GetSaveMetaInfo(fileIndex)->questItems & (1 << (item - 0x54))) != 0;
+    }
+
+    if (item >= ITEM_MEDALLION_FOREST && item <= ITEM_MEDALLION_LIGHT) {
+        return (Save_GetSaveMetaInfo(fileIndex)->questItems & (1 << (item - 0x66))) != 0;
+    }
+
+    if (item >= ITEM_KOKIRI_EMERALD && item <= ITEM_GERUDO_CARD) {
+        return (Save_GetSaveMetaInfo(fileIndex)->questItems & (1 << (item - 0x5A))) != 0;
+    }
+
+    if (item >= ITEM_SWORD_KOKIRI && item <= ITEM_SWORD_BGS) {
+        return (Save_GetSaveMetaInfo(fileIndex)->equipment & (1 << (item - 0x3B))) != 0;
+    }
+
+    if (item >= ITEM_SHIELD_DEKU && item <= ITEM_SHIELD_MIRROR) {
+        return (Save_GetSaveMetaInfo(fileIndex)->equipment & (1 << (item - 0x3E + 4))) != 0;
+    }
+
+    if (item >= ITEM_TUNIC_KOKIRI && item <= ITEM_TUNIC_ZORA) {
+        return (Save_GetSaveMetaInfo(fileIndex)->equipment & (1 << (item - 0x41 + 8))) != 0;
+    }
+
+    if (item >= ITEM_BOOTS_KOKIRI && item <= ITEM_BOOTS_HOVER) {
+        return (Save_GetSaveMetaInfo(fileIndex)->equipment & (1 << (item - 0x44 + 12))) != 0;
+    }
+
+    if (item == ITEM_SINGLE_MAGIC) {
+        return Save_GetSaveMetaInfo(fileIndex)->isMagicAcquired;
+    }
+
+    if (item == ITEM_DOUBLE_MAGIC) {
+        return Save_GetSaveMetaInfo(fileIndex)->isDoubleMagicAcquired;
+    }
+
+    if (item == ITEM_BRACELET) {
+        return ((Save_GetSaveMetaInfo(fileIndex)->upgrades & gUpgradeMasks[UPG_STRENGTH]) >> gUpgradeShifts[UPG_STRENGTH]) == 1;
+    }
+
+    if (item == ITEM_GAUNTLETS_SILVER) {
+        return ((Save_GetSaveMetaInfo(fileIndex)->upgrades & gUpgradeMasks[UPG_STRENGTH]) >> gUpgradeShifts[UPG_STRENGTH]) == 2;
+    }
+
+    if (item == ITEM_GAUNTLETS_GOLD) {
+        return ((Save_GetSaveMetaInfo(fileIndex)->upgrades & gUpgradeMasks[UPG_STRENGTH]) >> gUpgradeShifts[UPG_STRENGTH]) == 3;
+    }
+
+    if (item == ITEM_SCALE_SILVER) {
+        return ((Save_GetSaveMetaInfo(fileIndex)->upgrades & gUpgradeMasks[UPG_SCALE]) >> gUpgradeShifts[UPG_SCALE]) == 1;
+    }
+
+    if (item == ITEM_SCALE_GOLDEN) {
+        return ((Save_GetSaveMetaInfo(fileIndex)->upgrades & gUpgradeMasks[UPG_SCALE]) >> gUpgradeShifts[UPG_SCALE]) == 2;
+    }
+
+    if (item == ITEM_DOUBLE_DEFENSE) {
+        return Save_GetSaveMetaInfo(fileIndex)->isDoubleDefenseAcquired;
+    }
+
+    //greg
+    if (item == ITEM_RUPEE_GREEN) {
+        return Save_GetSaveMetaInfo(fileIndex)->gregFound;
+    }
+
+    return 0;
+}
+
+u8 ShouldRenderItem(s16 fileIndex, u8 item) {
+    //strength
+    if (item == ITEM_BRACELET && (HasItem(fileIndex, ITEM_GAUNTLETS_SILVER) != 0 || HasItem(fileIndex, ITEM_GAUNTLETS_GOLD) != 0)) {
+        return 0;
+    }
+
+    if (item == ITEM_GAUNTLETS_SILVER && HasItem(fileIndex, ITEM_GAUNTLETS_SILVER) == 0) {
```

Comment:
Should we be checking for !HasItem(fileIndex, ITEM_GAUNTLETS_GOLD) here?

Reply:
it looks like this is actually handled by `HasItem`, in that case, should the item bracelet above care about `(HasItem(fileIndex, ITEM_GAUNTLETS_SILVER) != 0 || HasItem(fileIndex, ITEM_GAUNTLETS_GOLD) != 0)`?

Reply:
Yes, because if you have silver or gold gauntlets, the bracelet icon should not render at all, not even dimmed.